### PR TITLE
Switch to using sphinx-js fork

### PR DESCRIPTION
**Pull request by hoodmane**
This allows us to remove a lot of code from sphinx_pyodide. It will take more effort to upstream the automodule documenter features, but all the type formatting features have been moved upstream and I added hooks to render xrefs into the types and to avoid the converter monkeypatch.

The new version also has some neat improvements like automatically fixing up `code between single upticks` into proper code instead of rendering it in italics, better "examples" rendering, etc. Mostly tiny things. Also we moved to typedoc 0.25.0 which itself does a lot of things a bit better.

### CODE REVIEW

**Comment by hoodmane** [comment]
@rth @ryanking13 Can someone review this when they get a chance?

File: src/js/tsdoc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "https://developer.microsoft.com/en-us/json-schemas/tsdoc/v0/tsdoc.schema.json",
+  "extends": ["typedoc/tsdoc.json"],
+  "noStandardTags": false,
+  "tagDefinitions": [
```
Comment (rth) [inline src/js/tsdoc.json:5]: Can you add a few comments about what these are used for,  for later reference?

**Review by rth — APPROVED** [review]
Thanks, @hoodmane, it's a nice simplification of the code sphinx_pyodide! I'll let you double-check that the rendering is working as expected )

File: src/js/tsdoc.json (hunk 1 above)
Comment (hoodmane) [reply src/js/tsdoc.json:5]: Well we can't add comments in json files...

**Comment by hoodmane** [comment]
> I'll let you double-check that the rendering is working as expected )

Yeah I spent a **lot** of time comparing the rendering to get all the details correct.

File: src/js/tsdoc.json (hunk 1 above)
Comment (rth) [reply src/js/tsdoc.json:5]: It shows that I haven't edited json for a while )

**Comment by rth** [comment]
Thanks a lot, Hood for your patience in doing this!

File: src/js/tsdoc.json (hunk 1 above)
Comment (hoodmane) [reply src/js/tsdoc.json:5]: I'll add a comment in about this file in `jsdoc.py`, because it is indeed easy to forget about it.

**Pull request merged**